cmd/bucket-stats.go

```diff
@@ -107,38 +234,203 @@ func (brs BucketReplicationStats) Clone() (c BucketReplicationStats) {
-	FailedSize int64 `json:"failedReplicationSize"`
-	// Total number of pending operations including metadata updates
-	PendingCount int64 `json:"pendingReplicationCount"`
-	// Total number of failed operations including metadata updates
-	FailedCount int64 `json:"failedReplicationCount"`
+	// Collect stats for failures
+	FailStats RTimedMetrics `json:"-"`
+
+	// Total number of failed operations including metadata updates in the last minute
+	Failed madmin.TimedErrStats `json:"failed"`
+	// Total number of completed operations
+	ReplicatedCount int64 `json:"replicationCount"`
 	// Replication latency information
 	Latency ReplicationLatency `json:"replicationLatency"`
 	// bandwidth limit for target
 	BandWidthLimitInBytesPerSecond int64 `json:"limitInBits"`
 	// current bandwidth reported
 	CurrentBandwidthInBytesPerSecond float64 `json:"currentBandwidth"`
+	// transfer rate for large uploads
+	XferRateLrg *XferStats `json:"-" msg:"lt"`
+	// transfer rate for small uploads
+	XferRateSml *XferStats `json:"-" msg:"st"`
+
+	// Deprecated fields
+	// Pending size in bytes
+	PendingSize int64 `json:"pendingReplicationSize"`
+	// Failed size in bytes
+	FailedSize int64 `json:"failedReplicationSize"`
+	// Total number of pending operations including metadata updates
+	PendingCount int64 `json:"pendingReplicationCount"`
+	// Total number of failed operations including metadata updates
+	FailedCount int64 `json:"failedReplicationCount"`
 }
 
 func (bs *BucketReplicationStat) hasReplicationUsage() bool {
-	return bs.FailedSize > 0 ||
+	return bs.FailStats.SinceUptime.Count > 0 ||
 		bs.ReplicatedSize > 0 ||
-		bs.ReplicaSize > 0 ||
-		bs.FailedCount > 0 ||
-		bs.PendingCount > 0 ||
-		bs.PendingSize > 0
+		bs.ReplicaSize > 0
+}
+
+func (bs *BucketReplicationStat) updateXferRate(sz int64, duration time.Duration) {
+	if sz > minLargeObjSize {
+		bs.XferRateLrg.addSize(sz, duration)
+	} else {
+		bs.XferRateSml.addSize(sz, duration)
+	}
+}
+
+func (bs *BucketReplicationStat) String() string {
+	s := fmt.Sprintf(" Transfer Rate Small Objects (Avg)=%f (Peak)=%f Current=%f", bs.XferRateSml.Avg, bs.XferRateSml.Peak, bs.XferRateSml.Curr)
+	return s
 }
 
 func (brs BucketReplicationStats) String() string {
 	s := "ReplicatedSize=" + fmt.Sprintf("%d", brs.ReplicatedSize) + "+\n ReplicaSize=" + fmt.Sprintf("%d", brs.ReplicaSize)
 	for arn, st := range brs.Stats {
-		s += "\n arn: " + arn + " ReplicatedSize=" + fmt.Sprintf("%d", st.ReplicatedSize) + "  +::ReplicaSize=" + fmt.Sprintf("%d", st.ReplicaSize)
+		s += "\n arn: " + arn + " ReplicatedSize=" + fmt.Sprintf("%d", st.ReplicatedSize) + "  +::ReplicaSize=" + fmt.Sprintf("%d", st.ReplicaSize) + fmt.Sprintf(" Transfer Rate Large Objects (Avg)=%f (Peak)=%f Current=%f", st.XferRateLrg.Avg, st.XferRateLrg.Peak, st.XferRateLrg.Curr) +
+			fmt.Sprintf(" Transfer Rate Small Objects (Avg)=%f (Peak)=%f Current=%f", st.XferRateSml.Avg, st.XferRateSml.Peak, st.XferRateSml.Curr) + st.FailStats.String()
 	}
 	return s
 }
```

Comment:
This is a longer string @poornas it is better to use `strings.Builder`

Reply:
also I wonder why we need a string() version of these structs - who is using them?

Reply:
this is just for debugging purposes - nobody uses this

Reply:
Can we just not have this code in there so that some-one may call it? comment it out saying ` /// debug code only`